Experiments/forest_QKS/qks.py

```diff
@@ -0,0 +1,223 @@
+    the ordering of the CNOT gates (before and after compilation they still do not match).
+    This is probably best just left up to the compiler.
+
+    For ease of reading the printed operations, the qubits are looped over several times.
+    """    
+    program = Program()
+    qubits = qc.qubits()
+    n_qubits = len(qubits)
+    var = 'theta'
+
+    ro = program.declare('ro', memory_type='BIT', memory_size=n_qubits)
+    thetas = {var+str(qubit):program.declare(var+str(qubit), memory_type = 'REAL') for qubit in qubits}
+
+    sq = int(np.sqrt(n_qubits))
+    lim = n_qubits-sq-1 
+    
+    for m in qubits: program += (RX(thetas[var+str(m)],m))
+    for m in qubits:
+        m_1 = m+1
+        m_sq = m+sq
+        skip = (m_1)%sq
+        
+        if m_1 < n_qubits:
+            if (m_sq >= n_qubits): program += CNOT(m,m_1)
+            else: program += CNOT(m,m_sq)
+    
+        if (m < lim) and (skip != 0): program += CNOT(m,m_1)
+
+    for m in qubits: program += MEASURE(m,ro[m])
+
+    print("program instructions from pyquil:\n")
+    for instruction in program.instructions:
+        print(instruction) 
+
+    return program
+
+
+def qks_run(qc, executable, theta):
+    theta_map = {'theta%s'%pos:[theta[pos]] for pos in range(len(theta))}
+    bitstrings = qc.run(executable, memory_map=theta_map)
+    avg_measurements = bitstrings.mean(axis=0)
+    return avg_measurements
+
+
+def logistic_regression(X_train, y_train, X_test, y_test):
+    lr = LogisticRegression(solver='lbfgs')
+    lr.fit(X_train, y_train)
+
+    train_acc = lr.score(X_train,y_train)
+    test_acc = lr.score(X_test,y_test)
+
+    print(
+        "accuracy\n----- \n training: {}\n test:     {}"
+          .format(train_acc, test_acc)
+         )
+
+    train_preds = lr.predict(X_train)
+    test_preds = lr.predict(X_test)
+
+    return train_preds, test_preds
+
+
+def make_plot(data, target, name, title):
+    plt.figure(figsize=(5,5))
+    plt.title(title)
+    plt.scatter(data[:,0], data[:,1], s=5, c=target)
+    plt.savefig(name)
+
+
+def main():
+    np.random.seed(1337)
+
+    n_episodes = 20    # number of episodes the dataset is iterated over
+    sigma2 = 1         # spread of the distribution.
+    n_trials = 1000    # number of shots
+    noisy = False      # use pre-configured noise model
+    q = 2              # number of qubits to use
+    plot = True        # plot graph of the classified datasets
+    train_size = 200   # 200 is default to get the size of the dataset in the paper
+    test_size = 50     # 50 is default to get the size of the dataset in the paper
+    
+    img_path = 'figs/'
+    num_cols = n_episodes*q # number of columns after QKS transformation
+
+    # make picture frames dataset
+    train, y_train = make_frames(train_size)
+    test, y_test = make_frames(test_size)
+
+    p = train.shape[-1] # dimension of input vector
+    r = 1 if p/q < 1 else int(p/q)  # number of elements that are non-zero s.t. r<=p
```

Comment:
This (ternary operator) is actually one of the few places where I think "one-lining" makes sense. 
1. Since the logic is super easy
2. Since ternary operators are common in different languages, so people are used to it.